src/snowcli/app/loggers.py

```diff
@@ -1,32 +1,136 @@
+import copy
 import logging
+import logging.config
+from pathlib import Path
+from typing import Any, Dict
 
 import typer
+from snowcli.api.config import (
+    get_logs_config,
+    is_default_logs_path,
+)
+from snowcli.api.exceptions import InvalidLogsConfiguration
 
-DEFAULT_LOG_FORMAT = "%(asctime)s %(levelname)s %(message)s"
-DEBUG_LOG_FORMAT = "%(asctime)s %(levelname)s [%(name)s] %(message)s"
+_DEFAULT_LOG_FILENAME = "snowcli.log"
+
+DEFAULT_LOGGING_CONFIG: Dict[str, Any] = {
+    "version": 1,
+    "disable_existing_loggers": True,
+    "formatters": {
+        "default_formatter": {
+            "class": "logging.Formatter",
+            "format": "%(asctime)s %(levelname)s %(message)s",
+            "datefmt": "%Y-%m-%d %H:%M:%S",
+        },
+        "detailed_formatter": {
+            "class": "logging.Formatter",
+            "format": "%(asctime)s %(levelname)s [%(name)s] %(message)s",
+            "datefmt": "%Y-%m-%d %H:%M:%S",
+        },
+    },
+    "filters": {},
+    "handlers": {
+        "console": {
+            "class": "logging.StreamHandler",
+            "formatter": "default_formatter",
+            "level": logging.ERROR,
+        },
+        "file": {
+            "class": "logging.handlers.TimedRotatingFileHandler",
+            "filename": None,
+            "when": "midnight",
+            "formatter": "detailed_formatter",
+            "level": logging.INFO,
```

Comment:
Do you need log level on handler level? Doesn't logger level overrides handler level? If that's true the logic probably can be simplified a bit

Reply:
Unfortunately. "Log level" works as a filter, simplified flow looks like this:
1. We call `logging.get_logger('snowcli.app.connector').info('message')`
2. Log message goes through loggers (`snowcli.app.connector` -> `snowcli.app` -> `snowcli` -> `root`) until it is filtered out by log level or finds one with handlers. (it is possible to send the message further after finding handlers with `propagate` setting)
3. Log message goes through all handlers and is printed by ones with appropriate level setting.
Details: https://docs.python.org/3/howto/logging.html#logging-flow 

So basically what I'm doing here is to set loggers to lowest possible level to pass through all interesting messages. 

Alternatively I could set loggers on `DEBUG` and let the handlers do the filtering. I didn't do that as for some reason someone took an effort to set level on `snowflake` and `snowcli` loggers, so I guessed there was some reasoning behind it. I might be wrong on this one tho, I'll rely on your opinion.